src/item-navigator.ts

```diff
@@ -0,0 +1,532 @@
+import { MetadataResponse } from '@internetarchive/search-service';
+import {
+  SharedResizeObserver,
+  SharedResizeObserverResizeHandlerInterface,
+} from '@internetarchive/shared-resize-observer';
+// @ts-ignore
+import { IAMenuSlider } from '@internetarchive/ia-menu-slider';
+import '@internetarchive/ia-menu-slider';
+
+import { ModalManagerInterface } from '@internetarchive/modal-manager';
+import '@internetarchive/icon-ellipses/icon-ellipses';
+import './loader';
+
+import {
+  IntManageSideMenuEvent,
+  IntSetOpenMenuEvent,
+  IntSetMenuContentsEvent,
+  IntSetMenuShortcutsEvent,
+  IntLoadingStateUpdatedEvent,
+  IntManageFullscreenEvent,
+} from './interfaces/event-interfaces';
+
+import { IntMenuProvider, IntMenuShortcut } from './interfaces/menu-interfaces';
+import './no-theater-available';
+@customElement('ia-item-navigator')
+export class ItemNavigator
+  extends LitElement
+  implements SharedResizeObserverResizeHandlerInterface {
+  @property({
+    type: Object,
+    converter: (value: string | MetadataResponse | null): MetadataResponse => {
+      if (value && typeof value === 'string') {
+        return new MetadataResponse(JSON.parse(atob(value)));
+      }
+      return value as MetadataResponse;
+    },
+  })
+  item: MetadataResponse | undefined = undefined;
+
+  @property({ type: String }) itemType = '';
+
+  @property({ type: String }) baseHost = 'archive.org';
+
+  @property({
+    converter: (arg: string | boolean | null) => {
+      if (typeof arg === 'boolean') {
+        return arg;
+      }
+      return arg === 'true';
+    },
+  })
+  signedIn = false;
+
+  @property({
+    type: Array,
+    hasChanged: (newVal: Array<object>, oldVal: Array<object>) => {
+      if (newVal !== oldVal) {
+        return true;
+      }
+      return false;
+    },
+  })
+  @property({ type: Array })
+  menuContents: IntMenuProvider[] = [];
+
+  @property({ type: Array }) menuShortcuts: IntMenuShortcut[] = [];
+
+  @property({ type: Boolean, reflect: true }) viewportInFullscreen:
+    | boolean
+    | null = null;
+
+  @property({ type: Boolean }) menuOpened = false;
+
+  @property({ type: String }) openMenu = '';
+
+  @property({ attribute: false }) modal?: ModalManagerInterface;
+
+  @property({ attribute: false })
+  sharedObserver?: SharedResizeObserver;
+
+  @property({ reflect: true, attribute: true }) loaded: boolean | null = null;
+
+  @state() openMenuState: 'overlay' | 'shift' = 'shift';
+
+  @query('#frame') private frame!: HTMLDivElement;
+
+  @query('slot[name="theater-header"]') private headerSlot!: HTMLDivElement;
+
+  disconnectedCallback() {
+    this.sharedObserver?.removeObserver({
+      handler: this,
+      target: this.frame,
+    });
+  }
+
+  firstUpdated(): void {
+    if (!this.modal) {
+      this.createModal();
+    }
+
+    this.startResizeObserver();
+  }
+
+  updated(changed: PropertyValues) {
+    if (changed.has('modal')) {
+      if (!this.modal) {
```

Comment:
Similar comment as the `sharedObserver`. I think this will prevent consumers from passing in a modal manager and I could see us doing that in offshoot since we have an app-wide one there.

Reply:
updated to receive only